pages/GSoC-21.md

```diff
@@ -0,0 +1,156 @@
+---
+layout: page
+title: Ideas page for GSoC 2021 | Hydraecosystem.org
+comments: true
+permalink: /gsoc-21
+---
+
+# Introduction
+Welcome to Hydra Ecosystem's ideas page for [GSoC 2021](https://summerofcode.withgoogle.com/). We would like to receive proposals that encompass different aspects of the development. Different tools are under development
+and candidates can present projects to fulfill multiple ideas on this page and issues listed at our Github [organization page](https://github.com/HTTP-APIs/).
+
+To get in touch with the Community:
+* [Slack](https://join.slack.com/t/hydraecosystem/shared_invite/enQtNzM3NTg5NzQ2MDUxLWU1MjM3ZGRhZWM4ZTg1ODBjMTljNTQwNzAwMGM3ZDlmYTY3Y2E4OGJmN2NlZWRjMWIzY2MzN2NjOTIyYmQ1ZjU)
+
+## Application Instructions
+- Students wanting to apply to the org must familarise themselves with the Github workflow.
+- Working on open issues is the best way for your skill to be recognised.
+- It is a good practice to search the history of issues for similar problems before creating a new issue on GitHub, this will avoid the creation of duplicated issues.
+- Try running our tools and going through the demos to get a better understanding of how things work.
+- Hang around on the Slack channel and talk to people about any problems or issues.
+- Make sure that any issue your create is verified before beginning to work on it.
+- Use the provided [template](https://docs.google.com/document/d/1qPR02o6jY4uFBdCf3S0ous4LG32YCe6ZJuDGojXH2JY/edit?usp=sharing) for writing your proposals.
+- Discuss your ideas with the mentors and get your proposals reviewed before submitting. [Note: Due to the large number of proposals we receive, we only review a proposal once per student, so make it count.]
+- A list of prerequisites and concepts needed to understand Hydra and the tools developed by Hydra Ecosystem can be found [here](https://www.hydraecosystem.org/Starting-Material).
+- If you have any doubts, ask questions in our super friendly Slack community. We'll try our best to answer all your questions.
+
+<div id="Ideas"></div>
+
+# [Ideas for GSoC 2021](#Ideas)
+
+## 1. General Improvements in [hydrus](https://github.com/HTTP-APIs/hydrus).
+### Key points:
+There is always room from improvement in our flagship server, hydrus:
+- Updating the server according to the Hydra Spec.
+- Creation of dynamic endpoints. Users need to be able to define endpoints that have some functionality. Right now hydrus can only serve static data and users are unable to define how incoming data is processed in a hydrus server.
+- General improvements in code and optimizing the server functionality, refactoring.
+- Keep up with the development of the querying system for the agent
+- Implement generalisation of Collection as a Resource.
+- Add more robust security features to hydrus.
+
```

Comment:
@sameshl please remove/modify the things we worked on during the last GSoC (point 2 and 5)